pkg/typesystem/typesystem.go

```diff
@@ -770,7 +770,7 @@ func hasEntrypoints(
 
 		loop := false
 		for _, child := range rw.Union.Child {
-			hasEntrypoints, childLoop, err := hasEntrypoints(typedefs, typeName, relationName, child, maps.Clone(visitedRelations))
```

Comment:
I need a little more time to think on this one, but I'm hesitant to drop the `maps.Clone` since it avoids sharing visited relations among children of the same immediate ancestor. I know for cycle detection this would be a deal-breaker, but I'm trying to comprehend a scenario where thay _may_ be an issue with entrypoint detection as well.

Can you think of any side-effects off of the top of your head?

Reply:
In the first line of `hasEntrypoints` we are already cloning the input parameter

Reply:
> In the first line of hasEntrypoints we are already cloning the input parameter

Right, but that cloned map is shared by all children of the inner union, exclusion, or intersection, which would be mutating each other's maps. That's the concern.

Reply:
🤔 I don't understand the concern